File: changwoomon/shortest_path/11265_끝나지않는파티.py
###### 11265번: 끝나지 않는 파티
# https://www.acmicpc.net/problem/11265
# 메모리/시간: 127152KB / 500ms (PyPy3)

import sys

input = sys.stdin.readline

N, M = map(int, input().split())

graph = [list(map(int, input().split())) for _ in range(N)]


for k in range(N):
    for i in range(N):
        for j in range(N):
            graph[i][j] = min(graph[i][j], graph[i][k] + graph[k][j])

for _ in range(M):
    A, B, C = map(int, input().split())
    A, B = A - 1, B - 1
    
    print("Enjoy other party") if graph[A][B] <= C else print("Stay here")

File: sally/shortest_path/11265_끝나지 않는 파티.cpp
/*
# DP
# Problem: 11265
# Memory: 3000KB
# Time: 40ms
*/
#include <iostream>
#include <cmath>
#include <string>
#include <string.h>
using namespace std;

#define success "Enjoy other party"
#define fail "Stay here"

int N, M;
int A, B, C;
int arr[501][501];

int main(void){
    cin.tie(NULL);
    ios::sync_with_stdio(false);

    cin >> N >> M;
    for(int i = 1; i <= N; i++)
        for(int j = 1; j <= N; j++)
            cin >> arr[i][j];

    for (int k = 1; k <= N; k++)
        for (int i = 1; i <= N; i++)
            for (int j = 1; j <= N; j++)
                arr[i][j] = min(arr[i][j], arr[i][k] + arr[k][j]);

    while(M--){
        cin >> A >> B >> C;
        if(arr[A][B] <= C) cout << success << '\n';
        else cout << fail << '\n';
    }

    return 0;
}